### PR TITLE
Permission fix

### PR DESCRIPTION
**Pull request by ghost**
@lsheiba Please review and merge into master.

* Fixes permission checks on license propose/accept functions
* Cleans up logging
* Removes authorized_id references since this hasn't been implemented and will cause confusion if left in for now.

### CODE REVIEW

**Review by lsheiba — COMMENTED** [review]
I was planning to disagree, but you done it yourself by reverting it in the next commit.

**Review by lsheiba — APPROVED** [review]
OK

File: nftsentry/src/license.rs [hunk 1]
```diff
@@ -99,13 +108,20 @@ impl Contract {
...
         let initial_storage_usage = env::storage_usage();
         
-        let master_id = env::predecessor_account_id();
+        let predecessor_id = env::predecessor_account_id();
+        let current_account_id = env::current_account_id();
+        let token = self.tokens_by_id.get(&token_id).expect("No token");
+
+
+        if predecessor_id != current_account_id {
+            panic!("Only the contract owner can propose a license update");
```
Comment (nmakhotkin) [inline nftsentry/src/license.rs]: current account ID is not always a contract owner, e.g. acc1 can be an owner of contract1 while acc1 != contract1. For this purpose `self.owner_id` needs to be used

File: nftsentry/src/license.rs [hunk 2]
```diff
@@ -9,20 +9,28 @@ const MIN_GAS_FOR_LICENSE_APPROVE_CALL: Gas = Gas(100_000_000_000_000);
...
+       
+        let predecessor_id = env::predecessor_account_id();
+        let current_account_id = env::current_account_id();
+        let token = self.tokens_by_id.get(&token_id).expect("No token");
 
-        let master_id = env::predecessor_account_id();
-    
-        self.internal_replace_license(&master_id,&token_id,&license);
+        if ( predecessor_id != current_account_id ) || ( predecessor_id != token.owner_id ) {
+            panic!("License can only be updated directly by the minter while the minter owns the token");
```
Comment (nmakhotkin) [inline nftsentry/src/license.rs]: predecessor (sender) can not be always == current_account_id (in that case the contract account should call itself). I guess here should be comparison against self.owner_id ?

File: nftsentry/src/license.rs (hunk 1 above)
Comment (ghost) [reply nftsentry/src/license.rs]: I'm removing the checks per the slack conversation.

File: nftsentry/src/license.rs (hunk 2 above)
Comment (ghost) [reply nftsentry/src/license.rs]: Removing per the slack convo

**Pull request merged**